pyro/ops/integrator.py

```diff
@@ -70,7 +70,16 @@ def potential_grad(potential_fn, z):
     z_keys, z_nodes = zip(*z.items())
     for node in z_nodes:
         node.requires_grad_(True)
-    potential_energy = potential_fn(z)
+    try:
+        potential_energy = potential_fn(z)
+    # deal with singular matrices
+    except RuntimeError as e:
+        if "singular U" in str(e):
```

Comment:
Is this the only way to detect if the matrix was singular? We'll need to change this check if the error message changes, though I think that's unlikely to.

Reply:
The error messages are:
```
inverse_cpu, U(4, 4) is zero, singular U
```
where `inverse` can be replaced by `cholesky`, `cpu` can be replaced by `gpu`. The general format is [at this line](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebraUtils.h#L170). So I pick the common one: singular U

> We'll need to change this check if the error message changes

I think the test will detect the new error message and we can update this check then.

Reply:
> I think the test will detect the new error message and we can update this check then.

Sounds good!